accepted/2023/net8.0-browser-tfm.md

```diff
@@ -0,0 +1,187 @@
+      {
+        ...
+      }
+  #else
+      public RecipesStore(HttpClient http)
+      {
+          _http = http;
+      }
+  #endif
+
+  #if BROWSER
+      public async Task<string> AddImage(Stream imageData)
+      {
+          var response = await _http.PostAsync("api/images", new StreamContent(imageData));
+          return await response.Content.ReadAsStringAsync();
+      }
+
+      public async Task<string> AddRecipe(Recipe recipe)
+      {
+          var response = await _http.PostAsJsonAsync("api/recipes", recipe);
+          return await response.Content.ReadAsStringAsync();
+      }
+  #else
+      public async Task<IEnumerable<Recipe>> GetRecipes(string? query)
+      {
+          ...
+      }
+
+      public Task<Recipe?> GetRecipe(string id)
+      {
+          ...
+      }
+
+      public Task<Recipe> UpdateRecipe(Recipe recipe)
+      {
+          ...
+      }
+
+      public Task<string> AddRecipe(Recipe recipe)
+      {
+          ...
+      }
+
+      public async Task<string> AddImage(Stream imageData)
+      {
+          ...
+      }
+
+      public Task<byte[]> GetImage(string filename)
+          => ...;
+  #endif
+  }
+  ```
+
+## Requirements
+
+Introduce a new net8.0-browser TFM that enables developers to author dlls with an API surface tailored for running in the browser, to leverage browser specific APIs, and to pivot their dependecies based on the platform.
+
+The net8.0-browser TFM will be a superset of net8.0, which means that an app or library targeting net8.0-browser will be able to consume net8.0 dependencies (for unsupported APIs we will continue using `[SupportedOSPlatform]`), however if a net8.0-browser version of the library is available, that would be preferred over the net8.0 version.
+
+The general rule will be that this TFM will work in the same way as net8.0-android, net8.0-ios, net8.0-windows, net8.0-maccatalyst, etc. do for Maui apps today.
+
+### Goals
+
+* Enable single project experiences for Blazor Webassembly applications that target ASP.NET Core environments and the browser with the same codebase.
+* Enable library authors to have dependencies thar are tailored for the browser.
+
+### Non-Goals
+
+**TL;DR**
+* Per browser TFM
+* Per browser version TFM
+* Browser specific API sets (although might be expanded in the future to cover this)
+
+#### Details
+
+It is well-known that the web ecosystem is a very large, diverse and changing landscape. There are multiple browsers with different sets of supported APIs and APIs are added and removed on a regular basis, which introduces risks for the stability of the platform.
+
+For these reasons, we are going to be very conservative about shipping additional APIs as part of the runtime and define a very restrictive criteria for when we think it is acceptable to add a new API. One such criteria can be:
+* The API is part of the current web standards.
+* All major vendors (Chrome, Firefox, Safari) ship an implementation of the API.
```

Comment:
NodeJS too ?

Reply:
Same comment as above, node is a different beast.